CMakeLists.txt

```diff
@@ -37,6 +40,23 @@ endif()
 # Dependencies
 # =============================================================================
 
+find_package(MPI)
+if (MPI_FOUND)
+    set(HDF5_PREFER_PARALLEL "ON")
+    add_definitions(-DHAVE_MPI)
+    include_directories (${MPI_INCLUDE_PATH})
+endif()
+
+find_package(HDF5)
+if (HDF5_FOUND)
+    if (HDF5_IS_PARALLEL)
```

Comment:
We are getting into combinatoric blowup realm here; the final PR will have to include updates to the testing to make sure all the versions are covered.  

Currently I'm seeing
MPI{ON/OFF} / HDF5{ON/OFF} / HDF5_IS_PARALLEL {ON/OFF} ...